receiver/awscontainerinsightreceiver/internal/stores/store.go

```diff
@@ -43,30 +43,32 @@ type K8sDecorator struct {
 	podStore *PodStore
 }
 
-func NewK8sDecorator(ctx context.Context, tagService bool, prefFullPodName bool, addFullPodNameMetricLabel bool, addContainerNameMetricLabel bool, includeEnhancedMetrics bool, logger *zap.Logger) (*K8sDecorator, error) {
+func NewK8sDecorator(ctx context.Context, tagService bool, prefFullPodName bool, addFullPodNameMetricLabel bool, addContainerNameMetricLabel bool, includeEnhancedMetrics bool, kubeConfigPath string, customHostIP string, customHostName string, logger *zap.Logger) (*K8sDecorator, error) {
 	hostIP := os.Getenv("HOST_IP")
 	if hostIP == "" {
-		return nil, errors.New("environment variable HOST_IP is not set in k8s deployment config")
+		hostIP = customHostIP
+		if hostIP == "" {
+			return nil, errors.New("environment variable HOST_IP is not set in k8s deployment config or passed as part of the agent config")
+		}
 	}
 
 	k := &K8sDecorator{
 		ctx:                         ctx,
 		addContainerNameMetricLabel: addContainerNameMetricLabel,
 	}
 
-	podstore, err := NewPodStore(hostIP, prefFullPodName, addFullPodNameMetricLabel, includeEnhancedMetrics, logger)
+	podstore, err := NewPodStore(hostIP, prefFullPodName, addFullPodNameMetricLabel, includeEnhancedMetrics, kubeConfigPath, customHostName, logger)
 	if err != nil {
 		return nil, err
 	}
 	k.podStore = podstore
 	k.stores = append(k.stores, podstore)
 
 	if tagService {
-		servicestore, err := NewServiceStore(logger)
-		if err != nil {
-			return nil, err
+		servicestore, err := NewServiceStore(kubeConfigPath, logger)
+		if err == nil {
```

Comment:
Why are we no longer erroring out?

Reply:
container insights failing (due to kubelet/api-server) failures should not affect the other functionalities of the agent